Automattic/wp-calypso
# Refactor Jetpack and WPCom product types

### Changes proposed in this Pull Request

This PR removes deprecated data and refactors types related to Jetpack products.

### Implementation notes

- Deleted the `records-details` folder, not imported anywhere anymore.
- Deleted `getJetpackProducts` in `packages/calypso-products/src/translations.js`, not used anywhere anymore.
- Added `PERIOD_LIST` to `packages/calypso-products/src/constants/terms.ts`.
- Move the `Product` type to `packages/calypso-products/src/types.ts`, and added more accurate annotations.
- Fixed the `Feature` type.

### Testing instructions

- Review code carefully, following the implementation notes along
- Download the PR and run Calypso
- Open the console
- Visit the plans page with both a simple and a Jetpack site
- Check that the page looks and behaves as in production, with a particular attention to the Site Search card
- Verify that you can complete a purchase
- Check that there's no relevant error in the console
